frontend/pages/model/[modelId]/release/[semver].tsx

```diff
@@ -29,24 +33,41 @@ export default function Release() {
     isReviewsLoading: isUserReviewsLoading,
     isReviewsError: isUserReviewsError,
   } = useGetReviewRequestsForUser()
+  const { model, isModelLoading, isModelError } = useGetModel(modelId, EntryKind.MODEL)
+  const { currentUser, isCurrentUserLoading, isCurrentUserError } = useGetCurrentUser()
 
-  const userCanReview =
-    reviews.filter((review) =>
-      userReviews.some(
-        (userReview) =>
-          userReview.model.id === review.model.id && userReview.accessRequestId === review.accessRequestId,
-      ),
-    ).length > 0
+  const currentUserRoles = useMemo(() => getCurrentUserRoles(model, currentUser), [model, currentUser])
+
+  const userCanReview = useMemo(
+    () =>
+      hasRole(currentUserRoles, ['msro', 'mtr']) &&
```

Comment:
I'm a little apprehensive about having authorisation now in both the frontend as well as the backend as it's duplication that will not be good for maintainability. Specifically here it's important to remember that in the near future that the roles for reviews will be dynamic and defined in the schema.

Reply:
Agreed. I was conscious of that when writing this and have made a mental note to prepare for the amount of added effort this work will add to dynamic roles. I've tried to limit that by having reusable logic, but unfortunately this is the best we can do for now.